### PR TITLE
Text::script should update existing script strings

### PR DESCRIPTION
**Pull request by Fedik**
Pull Request for Issue #41516 .

### Summary of Changes

The `HTMLDocument::setHeadData` restores `scriptOptions` correctly, hovewer any further call of `Text::script` override previously restored strings.

### Testing Instructions
Please follow #41516 .
Or add :
```php
Text::script('JYES');
$this->document->addScriptOptions('yes', 'Yes');
$this->document->getWebAssetManager()
    ->addInlineScript('console.log(Joomla.getOptions("yes", "nope") + " " + Joomla.Text._("JYES", "Nope"))', [], ['type' => 'module']);
```
After this line https://github.com/joomla/joomla-cms/blob/1e7527b60ec42032abe7bfb9c000bacf38f65845/components/com_content/src/View/Article/HtmlView.php#L242
Enable "Conservative caching"
Open any Article view.
Check browser console.

### Actual result BEFORE applying this Pull Request
First request: `Yes Yes`
All following requests: `Yes Nope`

### Expected result AFTER applying this Pull Request
Every reques there a log `Yes Yes`

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: 
- [x] No documentation changes for docs.joomla.org needed
- [ ] Pull Request link for manual.joomla.org: 
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 4.4-dev.

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 42b0d7aa43360b18e7bb4f32956b92ed0852ecfc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41543">issues.joomla.org/tracker/joomla-cms/41543</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41543">issues.joomla.org/tracker/joomla-cms/41543</a>.</sub>

File: libraries/src/Language/Text.php [hunk 1]
```diff
@@ -28,6 +28,8 @@ class Text
      *
      * @var    array
      * @since  1.7.0
+     *
+     * @deprecated 4.0 will be removed in 6.0. Without replacement.
```
Comment (laoneo) [inline libraries/src/Language/Text.php]: Deprecation's can only be done in minor releases.

**Review by laoneo — CHANGES_REQUESTED** [review]

File: libraries/src/Language/Text.php (hunk 1 above)
Comment (Fedik) [reply libraries/src/Language/Text.php]: How else it should be deprecated when PR stay open for months? ;)

**Review by laoneo — APPROVED** [review]

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**